correlation_modules/3_module_validation.py

```diff
@@ -317,258 +292,293 @@ def plot_dist_modules(clustering_method_list):
+pao1_operon_shuffle_prob.head()
 
-# +
-cumulative_distribution(
-    pa14_operon["Num_operon_modules"],
-    label="Number of modules containing operon genes",
-    color="red",
+# %%time
+pa14_operon_shuffle_prob = coverage_of_genesets(
+    pa14_membership_shuffle, pao1_operon, "operon"
 )
-cumulative_distribution(
-    pa14_operon["Num_random_modules"],
-    label="Number of modules containing random genes",
-    color="blue",
+pa14_operon_shuffle_prob.head()
+
+# %%time
+pao1_regulon_shuffle_prob = coverage_of_genesets(
+    pao1_membership_shuffle, pao1_regulon, "regulon"
 )
-_ = plt.legend()
-plt.title("Cumulative distribution of PA14 module counts (operon vs random genes)")
-plt.ylabel("Number of operons/random groups")
-plt.xlabel("The number of modules that operons/random genes are contained in")
+pao1_regulon_shuffle_prob.head()
 
-scipy.stats.ks_2samp(
-    pa14_operon["Num_operon_modules"], pa14_operon["Num_random_modules"]
+# %%time
+pa14_regulon_shuffle_prob = coverage_of_genesets(
+    pa14_membership_shuffle, pa14_regulon, "regulon"
 )
+pa14_regulon_shuffle_prob.head()
+
+# ## Plot distribution of probabilities
+#
+# Can we identify those operons, regulons that have high probability of being in the same module
 
 # +
-cumulative_distribution(
-    pao1_regulon["Num_regulon_modules"],
-    label="Number of modules containing regulon genes",
-    color="red",
-)
-cumulative_distribution(
-    pao1_regulon["Num_random_modules"],
-    label="Number of modules containing random genes",
-    color="blue",
+# Plot operon coverage
+# Note: We are only plotting the probabilities greater than 0 since there were many operons that had
+# a 0 probability, likely due to the small size of the operons
+# All probabilities for PA14 shuffled data is 0 which is why the plot is blank
+fig_operon, axes = plt.subplots(ncols=2, nrows=1, figsize=(10, 5))
+bins_shared = np.linspace(0, 1)
+
+fig_operon = sns.histplot(
+    pao1_operon_prob.loc[
+        pao1_operon_prob["pr(x,y in module|x,y in operon)"] > 0,
+        "pr(x,y in module|x,y in operon)",
+    ],
+    bins=bins_shared,
+    ax=axes[0],
+    label="true",
+)
+fig_operon = sns.histplot(
+    pa14_operon_prob.loc[
+        pa14_operon_prob["pr(x,y in module|x,y in operon)"] > 0,
+        "pr(x,y in module|x,y in operon)",
+    ],
+    bins=bins_shared,
+    ax=axes[1],
+    label="true",
+)
+fig_operon = sns.histplot(
+    pao1_operon_shuffle_prob.loc[
+        pao1_operon_shuffle_prob["pr(x,y in module|x,y in operon)"] > 0,
+        "pr(x,y in module|x,y in operon)",
+    ],
+    bins=bins_shared,
+    color="grey",
+    ax=axes[0],
+    label="shuffle",
+)
+fig_operon = sns.histplot(
+    pa14_operon_shuffle_prob.loc[
+        pa14_operon_shuffle_prob["pr(x,y in module|x,y in operon)"] > 0,
+        "pr(x,y in module|x,y in operon)",
+    ],
+    bins=bins_shared,
+    color="grey",
+    ax=axes[1],
+    label="shuffle",
```

Comment:
Isn't it weird that these probabilities are so low? Wouldn't you expect genes in the same operon to be really tightly coregulated?

Reply:
Looks like I didn't select the best matching module and operon pair, I was accidentally plotting all p-values for all mappings. I just fixed the code and there looks like there is more of a shift towards higher probabilities, though there are still some operons that have a fairly low probability of being in the same module. Based on the `describe` statistics, I don't see much that stands out. But I think its at least a good sign that most operons are clustered in the same module